### PR TITLE
Create parser_southafrica.toml

### PR DESCRIPTION
**Pull request by ekamau**
First draft of the South Africa data. Contains all tables. There are many comments - mostly questions / queries of things I wasn't sure how to map / include in the parser. Feedback and edits most welcome.

### CODE REVIEW

**Comment by abhidg** [comment]
Report on full data

|table       	|valid	|total	|percentage_valid|
|---------------|-------|-------|----------------|
|subject       	|493980	|494547	|99.885350% |
|visit         	|494143	|494547	|99.918309% |
|observation   	|2033898	|2038754	|99.761815% |

## subject

* 316: data must contain ['subject_id', 'country_iso3', 'enrolment_date', 'age', 'sex_at_birth', 'ethnicity', 'pathogen'] properties
* 251: data.age must be smaller than or equal to 120

## visit

* 404: data must contain ['visit_id', 'subject_id', 'country_iso3', 'start_date', 'outcome', 'date_outcome'] properties

## observation

* 4856: data.is_present must be boolean

File: isaric/parsers/datcov-southafrica.toml [hunk 1]
```diff
@@ -510,14 +510,15 @@
...
+[[observation]]
+  name = "history_of_fever"
+  phase = "followup"
+  date = { field = "flw2_survey_date_{n}" }
+  is_present = { field = "flw2_fever_{n}", values = { 0 = false, 1 = true } } # values = ['0', '1', '2']
+  if.any = [{ "flw2_fever_{n}" = 1 }, { "flw2_fever_{n}" = 0 }]
+  for.n.range = [1, 5]
 
-  # [[observation]]
+  # [[observation]]  # TODO: should this be merged into above?
```
Comment (ekamau) [inline isaric/parsers/datcov-southafrica.toml:521]: How would they be merged?

File: isaric/parsers/datcov-southafrica.toml [hunk 2]
```diff
@@ -510,14 +510,15 @@
...
-  #   name = "history_of_fever"
-  #   phase = "followup"
-  #   date = { ref = "followupDateHierarchy" }
-  #   is_present = { field = "flw2_fever", values = { 0 = false, 1 = true } } # values = ['0', '1', '2']
-  #   if = { flw2_fever = { "!=" = 2 } }
+[[observation]]
+  name = "history_of_fever"
+  phase = "followup"
+  date = { field = "flw2_survey_date_{n}" }
+  is_present = { field = "flw2_fever_{n}", values = { 0 = false, 1 = true } } # values = ['0', '1', '2']
```
Comment (ekamau) [inline isaric/parsers/datcov-southafrica.toml:517]: Clever .. I like this ..

File: isaric/parsers/datcov-southafrica.toml [hunk 3]
```diff
@@ -650,12 +653,13 @@
...
-  #   phase = "followup"
-  #   date = { ref = "followupDateHierarchy" }
-  #   is_present = { field = "flw2_headache", values = { 0 = false, 1 = true } }
-  #   if.any = [{ flw2_headache = 2 }, { flw2_headache = 4 }]
+[[observation]]
+  name = "headache"
+  phase = "followup"
+  date = { field = "flw2_survey_date_{n}" }
+  is_present = { field = "flw2_headache_{n}", values = { 0 = false, 1 = true } }
+  if.any = [{ "flw2_headache_{n}" = 1 }, { "flw2_headache_{n}" = 0 }]
```
Comment (ekamau) [inline isaric/parsers/datcov-southafrica.toml:661]: How does this part work? can't figure it out ... also applies for the other mappings above

**Review by ekamau — COMMENTED** [review]
Looks good. Clever idea to have the 'n' ..

**Pull request merged**